### PR TITLE
tool: split esp paths

### PR DESCRIPTION
**Pull request by nikstur**
To access paths on the ESP before or after installing generations, split EspPaths into general EspPaths that only depend on the path to the ESP and EspGenerationPaths which additionally depend on generation specific information (e.g. version number and initrd filename).

This refactoring is in preparation to a larger change to the way systemd-boot is installed.

### CODE REVIEW

File: rust/tool/src/esp.rs [hunk 1]
```diff
@@ -70,6 +56,38 @@ impl EspPaths {
     }
 }
 
+/// Contains the generation-specific paths to the ESP.
```
Comment (blitz) [inline rust/tool/src/esp.rs:59]: nit: I honestly had a bit of trouble making sense of this. Maybe:

```
/// Paths to the boot files of a specific generation.
///
/// All of these paths point into the ESP.
```

File: rust/tool/src/esp.rs [hunk 2]
```diff
@@ -5,62 +5,48 @@ use anyhow::{Context, Result};
 
 use crate::generation::Generation;
 
+/// Contains generation-independent paths on the ESP.
```
Comment (blitz) [inline rust/tool/src/esp.rs:8]: nit: Maybe:

```
/// Path to boot files that are not specific to a generation.
```

**Review by blitz — APPROVED** [review]
Looks pretty straightforward. Feel free to ignore the nits.

File: rust/tool/src/esp.rs (hunk 1 above)
Comment (nikstur) [reply rust/tool/src/esp.rs]: I still have a hard time finding a good comment/explanation. Your suggestions are definetely an improvement though.

**Pull request merged**